src/main/cocoa/CocoaWindow.mm

```diff
@@ -0,0 +1,731 @@
+
+                NSCursor* cursor = translate_cursor(pointer);
+                if (!cursor)
+                    return STATUS_UNKNOWN_ERR;
+
+                // Commit the cursor value
+                [cursor set]; // Sets the cursor immediately
+                enPointer = pointer;
+
+                return STATUS_OK;
+            }
+
+            mouse_pointer_t CocoaWindow::get_mouse_pointer()
+            {
+                return enPointer;
+            } 
+
+            status_t CocoaWindow::set_caption(const LSPString *caption)
+            {
+                if (caption == nullptr)
+                    return STATUS_BAD_ARGUMENTS;
+                if (pCocoaWindow == nullptr)
+                    return STATUS_BAD_STATE;
+
+                NSString *title = [NSString stringWithUTF8String:caption->get_utf8()];
+                if (!title)
+                    return STATUS_NO_MEM;
+
+                [pCocoaWindow setTitle:title];
+                return STATUS_OK;
+            }
+
+            status_t CocoaWindow::set_caption(const char *caption)
+            {
+                if (caption == nullptr)
+                    return STATUS_BAD_ARGUMENTS;
+
+                LSPString tmp;
+                return (tmp.set_utf8(caption)) ? set_caption(&tmp) : STATUS_NO_MEM;
+            }
+
+            status_t CocoaWindow::get_caption(LSPString *text)
+            {
+                if (text == nullptr)
+                    return STATUS_BAD_ARGUMENTS;
+                if (pCocoaWindow == nullptr)
+                    return STATUS_BAD_STATE;
+
+                NSString *title = [pCocoaWindow title];
+                const char *utf8 = [title UTF8String];
+                return (text->set_utf8(utf8)) ? STATUS_OK : STATUS_NO_MEM;
+            }
+
+            status_t CocoaWindow::get_caption(char *text, size_t len)
+            {
+                if (text == nullptr)
+                    return STATUS_BAD_ARGUMENTS;
+                if (len < 1)
+                    return STATUS_TOO_BIG;
+
+                LSPString tmp;
+                status_t res = get_caption(&tmp);
+                if (res != STATUS_OK)
+                    return res;
+
+                const char *utf8 = tmp.get_utf8();
+                size_t count = strlen(utf8) + 1;
+                if (len < count)
+                    return STATUS_TOO_BIG;
+
+                memcpy(text, utf8, count);
+                return STATUS_OK;
+            }
+
+            //TODO: Translate all Window Border Styles!
+            static NSWindowStyleMask get_ns_style(border_style_t style) {
+                using namespace lsp::ws;
+                switch (style) {
+                    case BS_DIALOG:         return NSWindowStyleMaskTitled;
+                    case BS_SINGLE:         return NSWindowStyleMaskTitled | NSWindowStyleMaskClosable | NSWindowStyleMaskMiniaturizable;
+                    case BS_NONE:           return NSWindowStyleMaskBorderless;
+                    /*
```

Comment:
The `BS_POPUP` and `BS_COMBO` border styles are important for implementing context menus and drop-down lists for combo boxes. I suppose they also should act as `BS_DROPDOWN` which is a borderless window.